examples/cpp/request_reply/CLIParser.hpp

```diff
@@ -0,0 +1,317 @@
+                print_help(EXIT_FAILURE);
+            }
+
+            consume_client_arguments(argc, argv, config);
+        }
+
+        return config;
+    }
+
+    /**
+     * @brief Parse the signal number into the signal name
+     *
+     * @param[in] signum signal number
+     *
+     * @return std::string signal name
+     */
+    static std::string parse_signal(
+            const int& signum)
+    {
+        switch (signum)
+        {
+            case SIGINT:
+                return "SIGINT";
+            case SIGTERM:
+                return "SIGTERM";
+#ifndef _WIN32
+            case SIGQUIT:
+                return "SIGQUIT";
+            case SIGHUP:
+                return "SIGHUP";
+#endif // _WIN32
+            default:
+                return "UNKNOWN SIGNAL";
+        }
+    }
+
+    /**
+     * @brief Parse the entity kind into std::string
+     *
+     * @param[in] entity entity kind
+     *
+     * @return std::string entity kind
+     */
+    static std::string parse_entity_kind(
+            const EntityKind& entity)
+    {
+        switch (entity)
+        {
+            case EntityKind::SERVER:
+                return "Server";
+            case EntityKind::CLIENT:
+                return "Client";
+            case EntityKind::UNDEFINED:
+            default:
+                return "Undefined entity";
+        }
+    }
+
+private:
+
+    /**
+     * @brief Consume the client arguments and store them in the config object
+     *
+     * @param[in] argc number of arguments
+     * @param[in] argv array of arguments
+     *
+     * @param[in,out] config config object to store the arguments
+     *
+     * @warning This method finishes the execution of the program if the input arguments are invalid
+     */
+    static void consume_client_arguments(
+            const int argc,
+            const char* const argv[],
+            config& config)
+    {
+        static_cast<void>(argc);
+        assert(argc >= 5);
+        config.x = consume_int16_argument(argv[2]);
+        config.operation = consume_operation_argument(argv[3]);
+        config.y = consume_int16_argument(argv[4]);
+    }
+
+    /**
+     * @brief Consume an int16 argument and return it
+     *
+     * @param[in] arg string argument to consume
+     *
+     * @return std::int16_t int16 argument
+     *
+     * @warning This method finishes the execution of the program if the input arguments are invalid
+     */
+    static std::int16_t consume_int16_argument(
+            const std::string& arg)
+    {
+        std::int16_t value = 0;
+
+        try
+        {
+            int input = std::stoi(arg);
+
+            if (input < std::numeric_limits<std::int16_t>::min() ||
+                    input > std::numeric_limits<std::int16_t>::max())
+            {
+                throw std::out_of_range("int16 argument out of range");
+            }
+
+            value = static_cast<std::int16_t>(input);
```

Comment:
Why are we using a `int16_t` and not a type whose range is bigger?

Reply:
I did not think it necessary